### DISCUSSION TITLE
Indent lost in big json file

### DISCUSSION THREAD

**Discussion by Lun4m** [lukas-reineke/indent-blankline.nvim#861] [Q&A] [answered]
I have a big json file, with 3 spaces indent. At the start it looks fine:
![Screenshot from 2024-04-18 15-18-52](https://github.com/lukas-reineke/indent-blankline.nvim/assets/15078179/aea61cab-4faa-45b9-bf69-f81b932843d3)

After some lines however, I get this:
![Screenshot from 2024-04-18 15-19-14](https://github.com/lukas-reineke/indent-blankline.nvim/assets/15078179/17bec1d9-b872-4885-a7d9-cef5675fe234)

It seems to depend on the fact I have `vim.opt.shfitwidth = 2`. 
Is it a bug or expected behavior?

Edit:
I'm also using [mini.indentscope](https://github.com/echasnovski/mini.indentscope) to highlight the indentation level at the cursor position, and it manages to detect the correct level.

**Comment by lukas-reineke** [comment, marked as answer]
This is expected behavior.
IBL uses both your settings, like `shfitwidth`, as well as the surrounding code, to determine indentations.
But looking at the surrounding code from the beginning of the file would be very slow, so by default it only looks at 30 lines above the first displayed line.
So if the start of the indentation is further up than that, it uses `shiftwidth` instead.

You can change how big this buffer is with `:help ibl.config.viewport_buffer.min`, but the correct solution is to just set `shiftwidth`.

**Reply by Lun4m** [reply]
Thanks for the explanation!

**Comment by Exagone313** [comment]
I have the same issue, I use nvim-treesitter for indentation, I have not defined any shiftwidth myself. Is the issue coming from treesitter?

EDIT: Added this config, but I think it is very weird that ibl has to deal with this, indentation level should come from treesitter.

```lua
require("ibl").setup{
	viewport_buffer = {
		min = 150,
	},
}
```
